### ISSUE TITLE
`fromPixels` returns incorrect values for `webgpu` backend

### ISSUE THREAD

**Issue by vladmandic** [tensorflow/tfjs#6851]
i've noticed some errors during model execution, so went back to check inputs and it seems that `fromPixels` returns incorrect values for `webgpu` backend

```js
await tf.setBackend('webgpu');
const buffer = await tf.browser.fromPixelsAsync(canvas); // input is a normal photo in 1280x720 resolution
const data = await buffer.data();
for (let idx = 0; idx < data.length; idx++) {
  if (data[idx] < 0 || data[idx] > 255) console.log({ idx, val: data[idx], length: data.length });
}
```

    {idx: 1094922, val: -1, length: 2764800}
    {idx: 1094925, val: -1, length: 2764800}
    {idx: 2275694, val: -1, length: 2764800}

i cannot think of any scenario where pixel value of '-1' is a valid thing to have, especially on only 3 random pixels out of entire (otherwise valid) image  

processing the same input shows no such problems when using `webgl` backend  

environment: tfjs 3.20.0, tfjs-backend-webgpu 0.0.1-alpha.13, chrome canary 107

**Comment by qjia7** [comment]
Assign to @xhcao, please take a look, thanks!

**Comment by xhcao** [comment]
@vladmandic , are there any steps that I could reproduce this issue locally? And comparing `webgl` backend with the same input, how many pixels that are different? Most or a few？

**Comment by xhcao** [comment]
@vladmandic Or you could get me the input data, record the canvas content(assume it is 2d context canvas) by getImageData() when there is -1 value after fromPixels.

**Comment by vladmandic** [comment]
digging in a bit into `tf.browser.fromPixelsAsync`...

- confirmed that it only happens for `webgpu` and not `webgl`
- issue only occurs when input is `HTMLVideoElement`
  - it DOES NOT occur for `HMTLCanvasElement`  
  - for example, copying same video frame to canvas and then using `fromPixelsAsync` on canvas shows no error
- issue is proportional to video complexity (*combination of scene change rate, video resolution and frame rate*)  
  - low complexity videos show no problems
  - medium ones may have 1-2 issue every few hundred frames
  - complex ones have hundreds of issues on each frame

basically, what happens is that after some offset within a frame, pixel values start drifting by `-1` in general  

so if actual pixel value happens to be `0`, it will result in `-1` being returned which is an invalid value.

all this leads me to believe this is a `webgpu` sync issue, but this pixel value drift after some offset is weird...

anyhow, full reproduction: <http://vladmandic.github.io/nudenet/src/verify.hmtl>  
and code is at <https://github.com/vladmandic/nudenet/blob/main/src/verify.ts>

**Comment by qjia7** [comment]
The above information reminds me that it may be a bug in [`importExternalTexture`](https://www.w3.org/TR/webgpu/#dom-gpudevice-importexternaltexture), which is a zero copy path for video.  @shaoboyan, who implemented this feature in chrome, any thoughts on this?

@vladmandic I suppose `tf.browser.fromPixels` has the similar issue. And suggest you switch to `fromPixels` since we plan to remove fromPixelsAsync in next big version upgrade. The `fromPixelsAsync` is just a simple wrapper of `fromPixels` to resolve a webgpu's old API problem, which doesn't exist now.

**Comment by vladmandic** [comment]
@shaoboyan live link i've posted (<https://vladmandic.github.io/nudenet/src/verify.html>, sorry there was a typo in the url) includes a sample video.
its only 10sec long, but on my system it generates thousands of errors per frame - and it logs to browser console.

**Comment by shaoboyan** [comment]
Thanks, I could reproduce this and I think this is due to the `GPUExternalTexture`.

**Comment by qjia7** [comment]
@shaoboyan Please file a chromium/dawn bug to trace the `GPUExternalTexture` issue. And this issue will be blocked by that one.
@vladmandic To continue to use `fromPixels` with video, you can set flag `WEBGPU_IMPORT_EXTERNAL_TEXTURE` to false. In this path, it will package the video to canvas and then call [`copyExternalImageToTexture()`](https://www.w3.org/TR/webgpu/#dom-gpuqueue-copyexternalimagetotexture) to do the following thing.

**Comment by shaoboyan** [comment]
Chromium issue filed: https://bugs.chromium.org/p/chromium/issues/detail?id=1368945. I'll handle this.

**Comment by xhcao** [comment]
As @qjia7 comment, Using `tf.env().set('WEBGPU_IMPORT_EXTERNAL_TEXTURE', false)` could work around this issue, but I did not verify whether there is a performance loss. Let's wait @shaoboyan to fix this issue, if the issue cannot be fixed in a short time, we could set `WEBGPU_IMPORT_EXTERNAL_TEXTURE` to false by default.

**Comment by shaoboyan** [comment]
@xhcao the reason why set `WEBGPU_IMPORT_EXTERNAL_TEXTURE` to false is not about this issue (Although it is the workaround way). The reason is that `GPUExternalTexture` in WebGPU will have some changes again, both in shader and api side. So we could disable this at this moment, and after spec related changes is ready, we could open it again.

**Comment by gyagp** [comment]
@vladmandic I temporarily disabled importExternalTexture() at https://github.com/tensorflow/tfjs/pull/6868, so you'll not encounter this issue with the new code. Anyway, @shaoboyan will investigate your issue and help to evolve the new spec and impl of importExternalTexture.

**Comment by vladmandic** [comment]
Sounds good - thanks!

**Comment by Varun-S10** [comment]
Hi @vladmandic, Thank you for your patience. We have tried to reproduce the issue. As a workaround, you can set tf.env().set('WEBGPU_IMPORT_EXTERNAL_TEXTURE', false) before initializing the backend. Could you please try the complete code above and let us know if it works for you?

**Comment by vladmandic** [comment]
i gave up on tfjs given that large number of my issues went unfixed for several years.
